### ISSUE TITLE
Plugin reload issue with shadow copy disabled

### ISSUE THREAD

**Issue by skoshelev** [nopSolutions/nopCommerce#6102]
nopCommerce version: 4.40+

Steps to reproduce the problem: 

Just to follow up on the scenario above when "UsePluginsShadowCopy" is disabled and you try to upload a plugin's package and don't have permissions to delete the dll file of the plugin.
If you got this error ""Access to the path 'Nop.Plugin.Feed.GoogleShopping.dll' is denied." in the administration, although the site seems to be working in the beginning,  actually the uploaded plugin's folder is corrupted.  All files and folders inside the plugin will be deleted i.e Views, Content, plugin.json etc. except the dll file that was left because of the access error above.

In other words if you use this functionality and you don't have permissions then this would corrupt the installation and you end up with broken plugins.

Maybe you should improve the upload plugin functionality by trying to delete the plugin's dll file first and if it is not successful then do not delete the rest of the files thus not corrupting the original plugin. If the dll file is deleted successfully then you can go ahead and delete the other files of the plugin.

See details on [this post](https://www.nopcommerce.com/en/boards/topic/92668/nopcommerce-450-suggestions-bug-reports-and-improvements/page/3#292278)

**Comment by skoshelev** [comment]
Closed #6102

**Comment by RomanovM** [comment]
@skoshelev  See [this comment](https://github.com/nopSolutions/nopCommerce/commit/ecb046566badfc11ce9aff438f9dbd8caf128698#commitcomment-68326274)

**Comment by skoshelev** [comment]
Closed #6102
